api/internal/plugins/loader/loader_test.go

```diff
@@ -78,3 +78,17 @@ func TestLoader(t *testing.T) {
 		}
 	}
 }
+
+func TestLoaderSetPluginConfigWorkingDir(t *testing.T) {
+	p := provider.NewDefaultDepProvider()
+	rmF := resmap.NewFactory(p.GetResourceFactory())
+	fsys := filesys.MakeFsInMemory()
+	c := types.EnabledPluginConfig(types.BploLoadFromFileSys)
+	pLdr := NewLoader(c, rmF, fsys)
+	pLdrCopy := *pLdr
+	pLdrCopy.DeepCopyPluginConfig()
+	pLdrCopy.SetPluginConfigWorkingDir("/tmp/dummy")
+	if pLdrCopy.Config().FnpLoadingOptions.WorkingDir != "/tmp/dummy" {
```

Comment:
I see you used `assert` in the integration test! If you use that here as well, these 3 lines can become 1. You can use `assert.Equal()` or `assert.Equalf()`. 

I also tend to use [the require package](https://pkg.go.dev/github.com/stretchr/testify/require), which is the same as `assert`, but ends the test on failure. Doesn't make a difference in your case, though.